source/tests/test_parallel_training.py

```diff
@@ -0,0 +1,35 @@
+import os
+import subprocess as sp
+import unittest
+
+from deepmd.cluster.local import get_gpus
+
+from common import tests_path
+
+
+class TestSingleMachine(unittest.TestCase):
+    def setUp(self):
+        try:
+            import horovod
+        except ImportError:
+            raise unittest.SkipTest("Package horovod is required for parallel-training tests.")
+        if len(get_gpus() or []) < 2:
```

Comment:
I think you also need to check whether tensorflow is built by CUDA? https://www.tensorflow.org/api_docs/python/tf/test/is_built_with_cuda

Reply:
Oh I think it should be not a problem. `get_gpus` should return empty if it's not built by CUDA.